src/AutoRest.CSharp/Mgmt/Output/Resource.cs

```diff
@@ -170,8 +170,8 @@ public Resource(IReadOnlyDictionary<OperationSet, IEnumerable<Operation>> allOpe
 
         public virtual Resource GetResource() => this;
 
-        //private string? _defaultName;
-        protected override string DefaultName => ResourceName;
+        // name after `{ResourceName}Resource`, unless the `ResourceName` already ends with `Resource`
+        protected override string DefaultName => ResourceName.AddResourceSuffixToResourceName();
```

Comment:
If we append the `Resource` suffix here, we might see the case when a resource name prepends to its child's resource name, the `Resource` suffix appears in the middle of a name.

The function calculating the resource name is here: https://github.com/Azure/autorest.csharp/blob/52af1748e404411abbc53504e5abc5149f76694a/src/AutoRest.CSharp/Mgmt/AutoRest/MgmtOutputLibrary.cs#L517
What do you think we first make sure the name returned by the above function does not end with the word `Resource` and append the `Resource` suffix here?
We do not need to worry the name between two resources will collide - which will happen if we do not do this in this way, and we have insurance on this because we will throw exceptions when two files are written with the same name

Reply:
We might need to ensure `ResourceName` never ends with the word `Resource`, and we can change here like this
```suggestion
        // name after `{ResourceName}Resource`, unless the `ResourceName` already ends with `Resource`
        protected override string DefaultName => ResourceName + _suffix; // where _suffix is `Resource`
```